### PR TITLE
NETOBSERV-306 : Enable ipv6 support

### PR DESCRIPTION
**Pull request by praveingk**
Adding the necessary changes to enable IPv6 support from both ebpf agent and user-space agent side.

### CODE REVIEW

**Comment by mariomac** [comment]
I got some linter errors when running `make lint` locally:

```
pkg/flow/record.go:56: File is not `goimports`-ed (goimports)

pkg/flow/record_test.go:56: File is not `goimports`-ed (goimports)
                0x06, 0x07, 0x08, 0x09, 0x06, 0x07, 0x08, 0x09, 0x06, 0x07, 0x08, 0x09, 0x06, 0x07, 0x08, 0x09,// network6: u8[16] src_ip
                0x0a, 0x0b, 0x0c, 0x0d, 0x0a, 0x0b, 0x0c, 0x0d, 0x0a, 0x0b, 0x0c, 0x0d, 0x0a, 0x0b, 0x0c, 0x0d,// network6: u8[16] dst_ip
pkg/exporter/grpc_proto.go:72: File is not `goimports`-ed (goimports)
                                        fr.NetworkV6.SrcAddr[13], fr.NetworkV6.SrcAddr[14], fr.NetworkV6.SrcAddr[15],}}},
```
I'll fix the repo to run linting automatically with unit tests.

**Comment by praveingk** [comment]
Sure Mario, Please let me know if I need to add anything else (tests) from my side.

**Review by mariomac — COMMENTED** [review]
Some minor styling suggestions.

File: bpf/flow.h [hunk 1]
```diff
@@ -17,13 +17,23 @@ struct data_link {
...
+struct v6ip {
+    struct in6_addr src_ip6;
+    struct in6_addr dst_ip6;
+} __attribute__((packed));
+
+struct network {
+    union {
+        struct v4ip v4ip;
+        struct v6ip v6ip;
+    }type;
```
[the remark this answers is not in the document]
Comment (praveingk) [reply bpf/flow.h]: Done

**Comment by praveingk** [comment]
@mariomac, I added in the latest commit to remove union in the data structure used for ringbuffer. Tested this one out extensively, and it does not have the truncation problem.

File: pkg/exporter/grpc_proto.go [hunk 2]
```diff
@@ -55,14 +55,53 @@ func (g *GRPCProto) ExportFlows(input <-chan []*flow.Record) {
 }
 
 func flowToPB(fr *flow.Record) *pbflow.Record {
+	if fr.Protocol == flow.IPv6Type {
```
[the remark this answers is not in the document]
Comment (praveingk) [reply pkg/exporter/grpc_proto.go]: Done

**Comment by mariomac** [comment]
Manually tested and worked. Let's wait for the checks to finish.

**Review by mariomac — APPROVED** [review]

**Pull request merged**